core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1153,7 +1153,16 @@ class DAGScheduler(
     }
 
     if (!stageIdToStage.contains(task.stageId)) {
-      // Skip all the actions if the stage has been cancelled.
+      logInfo("skip normal actions as stage cancelled")
+      // Need to handle tasks coming in late (speculative and jobs killed)
+      // post a task end event so accounting for things manually tracking tasks work.
+      // This really should be something other then success since the other speculative task
+      // finished first.
```

Comment:
I think that is fine for now since its combining failed vs success tasks.   I do think its a bit weird that Spark marks all speculative tasks as Success even when both obviously don't commit.  That is part of the other jira I was going to file though and if needed it can be split back apart at that point.

It does seem a  bit odd to throw SPARK-13054 in with the other changes in the same pr though.